libraries/node-core-library/src/FileSystem.ts

```diff
@@ -521,4 +812,42 @@ export class FileSystem {
   public static getRealPath(linkPath: string): string {
     return fsx.realpathSync(linkPath);
   }
+
+  /**
+   * An async version of {@link FileSystem.getRealPath}.
+   */
+  public static getRealPathAsync(linkPath: string): Promise<string> {
+    return fsx.realpath(linkPath);
+  }
+
+  // ===============
+  // UTILITY FUNCTIONS
+  // ===============
+
+  /**
+   * Returns true if the error provided indicates the file or folder
+   * does not exist.
+   */
+  public static isNotExistError(error: NodeJS.ErrnoException): boolean {
+    return error.code === 'ENOENT' || error.code === 'ENOTDIR';
+  }
+
+  /**
+   * Re-throws the provided error if the error indicates the file or folder
+   * does not exist.
+   */
+  public static throwIfIsNotExistError(error: NodeJS.ErrnoException): void {
```

Comment:
Extracting this into a function does not seem very useful. It's a tiny bit of code that is arguably more readable inline. That also eliminates an unhelpful call stack record.